src/app-router.js

```diff
@@ -256,7 +264,11 @@ function resolveInstruction(instruction, result, isInnerInstruction, router) {
 function restorePreviousLocation(router) {
   let previousLocation = router.history.previousLocation;
   if (previousLocation) {
-    router.navigate(router.history.previousLocation, { trigger: false, replace: true });
+    Promise.resolve().then(() => {
```

Comment:
Why promise.resolve? I could see this causing breaking changes in someone's app because it fundamentally changes the timing.

Reply:
https://github.com/aurelia/router/issues/528#issuecomment-409493196

Reply:
I had some issues in first version (that might've been unrelated), but also that I thought the part that it was replacing, `router.navigate`, eventually lead to a promise. I'll remove it for the next version.